File: EduLink.Server/Object/BaseObject.cs
namespace webapi.Object
{
    using Microsoft.EntityFrameworkCore;
    using System;
    using System.ComponentModel.DataAnnotations;
    using System.ComponentModel.DataAnnotations.Schema;

    [PrimaryKey(nameof(Id))]
    public abstract class BaseObject
    {
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int Id { get; set; }

        public DateTime CreatedDate { get; set; }

        public DateTime? LastUpdateDate { get; set; }

        public BaseObject()
        {
        }
    }
}

File: edulink.client/src/app/models/BaseObject.ts
export abstract class BaseObject {
  id: number = 0;
  createdDate: Date = new Date;
  lastUpdateDate?: Date;

  //constructor(id?: number, createdDate?: Date, lastUpdateDate?: Date) {
  //  if (id !== undefined) {
  //    this.id = id;
  //  }
  //  if (createdDate !== undefined) {
  //    this.createdDate = createdDate;
  //  }
  //  if (lastUpdateDate !== undefined) {
  //    this.lastUpdateDate = lastUpdateDate;
  //  }
//}
}
